### PR TITLE
[Verify] Update new-architecture input fields on base screens to pass through all textfield props

### PR DESCRIPTION
**Pull request by manojleaton**
Fixes #BLUI-4302 .

#### Changes proposed in this Pull Request:

- Added TextFieldProps type in setPassword, VerifyCodeScreenBase, ChangePasswordDialog and ForgotPasswordScreen.

#### To Test:

- yarn start:example

#### Any specific feedback you are looking for?

-

### CODE REVIEW

File: login-workflow/src/components/ChangePasswordDialog/ChangePasswordDialogBase.tsx [hunk 1]
```diff
@@ -99,6 +100,7 @@ export const ChangePasswordDialogBase: React.FC<ChangePasswordDialogProps> = (pr
                         id="current-password"
                         label={currentPasswordLabel}
                         value={currentPassword}
+                        {...currentPasswordTextFieldProps}
```
Comment (ektaghag-eaton) [inline login-workflow/src/components/ChangePasswordDialog/ChangePasswordDialogBase.tsx:103]: Do we need to merge OnPasswordChange here like https://github.com/etn-ccis/blui-react-workflows/blob/dev/login-workflow/src/screens/AccountDetailsScreen/AccountDetailsScreen.tsx#L142.

Same comment for other screens

File: login-workflow/src/components/SetPassword/types.ts [hunk 2]
```diff
@@ -71,4 +72,10 @@ export type SetPasswordProps = {
      * @returns void
      */
     onSubmit?: () => void;
+
```
Comment (JeffGreiner-eaton) [inline login-workflow/src/components/SetPassword/types.ts:75]: you can refer to account details screen props for this but we've been using this for docs. But update everywhere where TextFieldProps will be used

/**
         * The props to pass to the password field.
         * See [MUI's TextFieldProps API](https://mui.com/material-ui/api/text-field/) for more details.
         */
        passwordTextFieldProps?: TextFieldProps;

**Comment by huayunh** [comment]
And ensure you update docs too, including the readmes and jsdocs

File: login-workflow/src/components/SetPassword/SetPassword.tsx [hunk 3]
```diff
@@ -84,13 +86,15 @@ export const SetPassword: React.FC<React.PropsWithChildren<SetPasswordProps>> =
                 inputRef={passwordRef}
                 label={newPasswordLabel}
                 value={passwordInput}
+                error={shouldValidatePassword && !isValidPassword()}
+                sx={TextFieldStyles(theme)}
+                {...passwordTextFieldProps}
```
Comment (huayunh) [inline login-workflow/src/components/SetPassword/SetPassword.tsx:91]: Any particular reason the `passwordTextFieldProps` is spreaded here as opposed to spreaded towards the end of this props list? I'd like to hear your reasoning.

Comment (huayunh) [reply login-workflow/src/components/SetPassword/SetPassword.tsx:91]: Same for the other spreads you added in this pull request

Comment (manojleaton) [reply login-workflow/src/components/SetPassword/SetPassword.tsx:91]: To override the defaults which we set in the base screen components, spreaded the `TextFieldProps` towards the end of the props list.

**Review by surajeaton — COMMENTED** [review]
Please resolve the conflicts.

File: login-workflow/src/components/SetPassword/SetPassword.tsx (hunk 3 above)
Comment (huayunh) [reply login-workflow/src/components/SetPassword/SetPassword.tsx:93]: I do not understand. I see that you still have `onChange`, `onKeyUp` and `onBlur` under the `{...passwordTextFieldProps}`, so this is not the end of the props list of `PasswordTextField`?

Comment (manojleaton) [reply login-workflow/src/components/SetPassword/SetPassword.tsx:93]: Just had a discussion with Joe in Office hours regarding this, so we need to merge events with `TextFieldProps` like we did for `WorkFlowActions`.

File: login-workflow/src/components/SetPassword/SetPassword.tsx [hunk 4]
```diff
@@ -84,14 +88,24 @@ export const SetPassword: React.FC<React.PropsWithChildren<SetPasswordProps>> =
                 inputRef={passwordRef}
                 label={newPasswordLabel}
                 value={passwordInput}
-                onChange={(evt: ChangeEvent<HTMLInputElement>): void => onPassChange(evt.target.value)}
+                error={shouldValidatePassword && !isValidPassword()}
+                sx={TextFieldStyles(theme)}
+                {...passwordTextFieldProps}
+                onChange={(evt: ChangeEvent<HTMLInputElement>): void => {
+                    // eslint-disable-next-line no-unused-expressions
+                    passwordTextFieldProps?.onChange && passwordTextFieldProps.onChange(evt);
```
Comment (ektaghag-eaton) [inline login-workflow/src/components/SetPassword/SetPassword.tsx:96]: This line is not helpful as we do nothing when this expression is true or false. Can you please elaborate more on why we are using this?

Comment (manojleaton) [reply login-workflow/src/components/SetPassword/SetPassword.tsx:96]: We are checking for the onChange event passed or not in passwordTextFieldProps, if it's false we are not going to call that callBack function.

**Review by surajeaton — COMMENTED** [review]
Please resolve the conflicts.

File: login-workflow/src/components/SetPassword/SetPassword.tsx [hunk 5]
```diff
@@ -84,14 +88,24 @@ export const SetPassword: React.FC<React.PropsWithChildren<SetPasswordProps>> =
                 inputRef={passwordRef}
                 label={newPasswordLabel}
                 value={passwordInput}
-                onChange={(evt: ChangeEvent<HTMLInputElement>): void => onPassChange(evt.target.value)}
+                error={shouldValidatePassword && !isValidPassword()}
+                sx={TextFieldStyles(theme)}
+                {...passwordTextFieldProps}
```
Comment (JeffGreiner-eaton) [inline login-workflow/src/components/SetPassword/SetPassword.tsx:93]: just so I know, why do we have both of these? just for theme aware items? plus additional sx styles?
 sx={TextFieldStyles(theme)}
                {...passwordTextFieldProps}

**Comment by JeffGreiner-eaton** [comment]
Just putting this here so we all know...

Confused when running tests locally with yarn test and running the one test file.

locally when running yarn test I see failed test here...

image
image

and not sure about the findByRole because the output from the test only finds these roles.

image
image
image


when I run only the test file itself the test passes and also passes in circleCI.

**Review by JeffGreiner-eaton — APPROVED** [review]

**Pull request merged**